lib/jekyll-admin/server/page.rb

```diff
@@ -32,21 +34,79 @@ class Server < Sinatra::Base
 
       private
 
+      def request_path
+        sanitized_path File.join(site.source, request_payload["path"])
+      end
+
+      def filename
+        "#{params["path"]}.#{params["ext"]}"
+      end
+
       def pages
         site.pages.select(&:html?)
       end
 
+      def directory_pages
+        pages.find_all do |p|
+          File.dirname(File.join(site.source, p.path)) == directory_path
```

Comment:
I'd use `sanitized_path` rather than `File.join` here to prevent directory traversal.